lib/ui/painting/image_descriptor.cc

```diff
@@ -103,27 +72,28 @@ void ImageDescriptor::initEncoded(Dart_NativeArguments args) {
     return;
   }
 
-  // This call will succeed if Skia has a built-in codec for this.
-  // If it fails, we will check if the platform knows how to decode this image.
-  std::unique_ptr<SkCodec> codec =
-      SkCodec::MakeFromData(immutable_buffer->data());
-  fml::RefPtr<ImageDescriptor> descriptor;
-  if (!codec) {
-    std::unique_ptr<SkImageGenerator> generator =
-        PLATFORM_IMAGE_GENERATOR(immutable_buffer->data());
-    if (!generator) {
-      // We don't have a Skia codec for this image, and the platform doesn't
-      // know how to decode it.
-      Dart_SetReturnValue(args, tonic::ToDart("Invalid image data"));
-      return;
-    }
-    descriptor = fml::MakeRefCounted<ImageDescriptor>(immutable_buffer->data(),
-                                                      std::move(generator));
-  } else {
-    descriptor = fml::MakeRefCounted<ImageDescriptor>(immutable_buffer->data(),
-                                                      std::move(codec));
+  // This has to be valid because this method is called from Dart.
+  auto dart_state = UIDartState::Current();
+  auto registry = dart_state->GetImageGeneratorRegistry();
+
+  if (!registry) {
+    Dart_SetReturnValue(
+        args, tonic::ToDart("Image generator registry not available"));
```

Comment:
Good question. This was initially copying the way ImageDecoder is [checked](https://github.com/flutter/engine/blob/master/lib/ui/painting/single_frame_codec.cc#L52-L56); I think the only real option would be to file a bug in this situation since it should get propagated to all isolates. What would you think if I appended something like: "This should never happen, please file a bug"?

Reply:
Perhaps something like:

> Flutter platform specific code failed to register image decoders. If you are writing your own platform embedding, use the Foo function to register these. Otherwise, please file a bug on https://github.com/flutter/flutter/issues."

?

Reply:
Thanks, I made the errors more actionable. Note that the registry itself is provided by Engine, so I think embedders have no say in messing this particular piece up. If no decoders are registered, it'll just behave like it can't recognize any images.

But I definitely think logging a warning when no decoders are installed is going to save someone a lot of time down the road, so I went ahead and added one to the registry lookup. I still need to make it easier for embedder implementers to register new decoders, so the instructions will change. Lemme know what you think.